packages/@react-native-windows/cli/src/generator-windows/index.ts

```diff
@@ -179,6 +180,12 @@ export async function copyProjectTemplateAndReplace(
   const packageGuid = uuid.v4();
   const currentUser = username.sync()!; // Gets the current username depending on the platform.
 
+  let mainComponentName = newProjectName;
+  const appJsonPath = findUp.sync('app.json', {cwd: destPath});
```

Comment:
We should use the async version of this if possible.

Reply:
Same below. File system reads can take arbitrarily long (e.g. over a network file system) and can cause the spinner to stutter.

Reply:
Switched to using async for findup, but we literally do every fs fead/write sync in template generation.

Reply:
There's some history behind that. Up until 0.63 the generator boundary was synchronous, so everything in it _needed_ to be sync. That was fixed to enable usage of standard async prompting libraries. In the future it would be great to use methods like `fs.promises.readFile` instead of `fs.readFileSync` to avoid more debt.

Reply:
I tried making some more of the operations in autolinking async but IIRC the entrypoint for autolinking _must_ be synchronous

Reply:
I remember for generator I had to change a chain of like 10 functions to be async. I think the CLI allows async commands though.

We have ESLint rules now that warn if you're missing an "await", so that style of refactoring should be pretty safe.

Reply:
IIRC when I made the autolink entrypoint async, the RN CLI did not await it
See https://github.com/microsoft/react-native-windows/pull/7196

Reply:
It looks like commands can be async, but dependencyConfig/projectConfig cannot be. See https://github.com/react-native-community/cli/blob/master/packages/cli-types/src/index.ts

Super nit, but we should probably be setting an explicit type for our dependency functions so the type checker can catch if we have unexpected return type for those functions (eg sync vs async). The react-native-config.js file exposing this I think has to be untyped, so we need to check types within the boundary.